### PR TITLE
[DPE-2717] Copy dashboard changes from K8s and use the correct topology dispatcher

### PR DESCRIPTION
**Pull request by dragomirp**
* Copy over dash changes from the K8s charm.
* Use the correct custom event dispatcher depending on the juju version

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
       "groupName": "Juju 2"
     }, {
       "matchPackageNames": ["urllib3"],
-      "allowedVersions": "<2.0.0"
+      "allowedVersions": "<2.1.0"
```
Comment (dragomirp) [inline renovate.json:46]: Boto moved the pin

File: src/charm.py [hunk 2]
```diff
@@ -99,7 +99,12 @@ def __init__(self, *args):
 
         self.secrets = {APP_SCOPE: {}, UNIT_SCOPE: {}}
 
-        self._observer = ClusterTopologyObserver(self)
+        juju_version = JujuVersion.from_environ()
+        if juju_version.major > 2:
+            run_cmd = "/usr/bin/juju-exec"
```
Comment (dragomirp) [inline src/charm.py:104]: The command name changed in juju 3

File: renovate.json [hunk 3]
```diff
@@ -41,9 +41,6 @@
       "matchDatasources": ["pypi"],
       "allowedVersions": "<3.0.0",
       "groupName": "Juju 2"
-    }, {
-      "matchPackageNames": ["urllib3"],
-      "allowedVersions": "<2.0.0"
```
Comment (dragomirp) [inline renovate.json:46]: Botocore moved the pin

**Review by marceloneppel — APPROVED** [review]
LGTM! Thanks!

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**